src/coreclr/scripts/superpmi-setup.py

```diff
@@ -0,0 +1,395 @@
+            shutil.copy2(src_item, dst_item)
+
+
+def copy_files(src_path, dst_path, file_names):
+    """Copy files from 'file_names' list from 'src_path' to 'dst_path'.
+    It retains the original directory structure of src_path.
+
+    Args:
+        src_path (string): Source directory from where files are copied.
+        dst_path (string): Destination directory where files to be copied.
+        file_names ([string]): List of full path file names to be copied.
+    """
+
+    print('### Copying below files to {0}:'.format(dst_path))
+    print('')
+    print(file_names)
+    for f in file_names:
+        # Create same structure in dst so we don't clobber same files names present in different directories
+        dst_path_of_file = f.replace(src_path, dst_path)
+
+        dst_directory = path.dirname(dst_path_of_file)
+        if not os.path.exists(dst_directory):
+            os.makedirs(dst_directory)
+        shutil.copy2(f, dst_path_of_file)
+
+
+def partition_files(src_directory, dst_directory, max_size, exclude_directories=[], exclude_files=native_binaries_to_ignore):
+    """ Copy bucketized files based on size to destination folder.
+
+    Args:
+        src_directory (string): Source folder containing files to be copied.
+        dst_directory (string): Destination folder where files should be copied.
+        max_size (int): Maximum partition size in bytes
+        exclude_directories ([string]): List of folder names to be excluded.
+        exclude_files ([string]): List of files names to be excluded.
+    """
+
+    sorted_by_size = get_files_sorted_by_size(src_directory, exclude_directories, exclude_files)
+    partitions = first_fit(sorted_by_size, max_size)
+
+    index = 0
+    for p_index in partitions:
+        file_names = [curr_file[0] for curr_file in partitions[p_index]]
+        curr_dst_path = path.join(dst_directory, str(index), "binaries")
+        copy_files(src_directory, curr_dst_path, file_names)
+        index += 1
+
+
+def set_pipeline_variable(name, value):
+    """ This method sets pipeline variable.
+
+    Args:
+        name (string): Name of the variable.
+        value (string): Value of the variable.
+    """
+    define_variable_format = "##vso[task.setvariable variable={0}]{1}"
+    print("{0} -> {1}".format(name, value)) # logging
+    print(define_variable_format.format(name, value)) # set variable
+
+
+def main(main_args):
+    """ Main entrypoint
+
+    Args:
+        main_args ([type]): Arguments to the script
+    """
+    coreclr_args = setup_args(main_args)
+    source_directory = coreclr_args.source_directory
+
+    # CorelationPayload directories
+    correlation_payload_directory = path.join(coreclr_args.source_directory, "payload")
+    superpmi_src_directory = path.join(source_directory, 'src', 'coreclr', 'scripts')
+    superpmi_dst_directory = path.join(correlation_payload_directory, "superpmi")
+    arch = coreclr_args.arch
+    helix_source_prefix = "official"
```

Comment:
What does "official" mean?

Reply:
I am not entirely sure. I believe we need to set the `HelixSource` to a value that starts with `official`.